### DISCUSSION TITLE
What is `use Rex -base`?

### DISCUSSION THREAD

**Discussion by rwp0** [RexOps/Rex#1544] [Questions and answers] [answered]
I see this import in modules generated with `rexify <module> --create-module`.

I didn't find a documentation for it looking up in [pod](https://metacpan.org/pod/Rex#FEATURE-FLAGS).

I usually replace the line with: `use Rex -feature => [ '1.4' ];` for using the latest features.

If an alias, how does it differ from the 1.4 feature flag?

Thanks.

**Comment by ferki** [comment, marked as answer]
`-base`, `-feature` and `-minimal` controls what happens during compilation time when `use Rex ...` is encountered. It boils down to a set of imports, and in case of `-feature` also handling the configuration options according to the features flags passed to it.

The difference is roughly this:

- [`-minimal`](https://github.com/RexOps/Rex/blob/ddb42aed1386aa68c276ae28d2daedbac7a7a7ea/lib/Rex.pm#L563-L569) is a minimal set of imports. It's useful for module development to avoid pulling in things that might not strictly required by the module's functionality.
- [`-base`](https://github.com/RexOps/Rex/blob/ddb42aed1386aa68c276ae28d2daedbac7a7a7ea/lib/Rex.pm#L571-L631) is a set of imports that is deemed necessary for the intended out-of-the-box Rex experience. It's useful in modules to match the same set of imports as the currently installed Rex version intends to provide.
- [`-feature`](https://github.com/RexOps/Rex/blob/ddb42aed1386aa68c276ae28d2daedbac7a7a7ea/lib/Rex.pm#L633-L941) imports the shared `-base` set, and then handles any configuration options according to the list of feature flags passed to it. It's mostly useful in the Rexfile to describe the desired features to be (de)activated for the given solution as a whole. It's usually less useful in modules, as it might unintentionally override the desired feature set described in the Rexfile.
